### PR TITLE
[14.0][IMP] stock_barcodes: some improvements

### PR DESCRIPTION
**Pull request by dalonsod**
Backport of #495

### CODE REVIEW

**Comment by dalonsod** [comment]
@sergio-teruel could you review?

Regarding to commit history, I've tried my best trying to preserve it when porting commits and later making required adjustments for v14.

Thanks in advance!!

**Review by elvise — APPROVED** [review]
Functional 👍

**Review by sergio-teruel — APPROVED** [review]

File: stock_barcodes/wizard/stock_barcodes_read_inventory.py [hunk 1]
```diff
@@ -1,93 +1,95 @@
...
-                ),
+    def _compute_inventory_quant_ids(self):
+        for wiz in self:
+            quants = self.env["stock.quant"].search(
+                [
+                    ("user_id", "=", self.env.user.id),
+                    "|",
+                    ("inventory_quantity_set", "=", True),
+                    ("inventory_date", "<=", fields.Date.context_today(self)),
+                ],
```
Comment (alexis-via) [inline stock_barcodes/wizard/stock_barcodes_read_inventory.py:29]: These 3 fields user_id, inventory_quantity_set and inventory_date don't exist on stock.quant on odoo v14 !!! These fields exists in v15 but not in v14.

**Comment by alexis-via** [comment]
In fact, it's not just a matter of those 3 fields: in v15, stock.inventory and stock.inventory.line have been removed, so there are major differences between an inventory in v15 and an inventory in v14. In this PR, the code has been backported without change from v15 to v14 for the inventory ; I think we'll have to completely roll-back the inventory part of this PR.

**Comment by dalonsod** [comment]
Yes, we know it... we'll make PR today or tomorrow, if we don't find more problems than it. Thanks!!

**Pull request merged**